doc/tutorial/search/index.md

```diff
@@ -0,0 +1,257 @@
+The general solution is to use a data structure called index. An index is built over the reference and only needs to be computed once for a given data set. It is used to speed up the search in the reference and can be re-used by storing it to disk and loading it again without recomputation.<br>
+There are two groups of indices: hash-tables and suffix-based indices, SeqAn3 implements the FM-Index and Bi-FM-Index as suffix-based indices. While the Bi-FM-Index needs more space (a factor of about 2), it allows faster queries.<br>
+Given an index, SeqAn3 will choose the best search strategy for you. Since very different algorithms may be selected internally depending on the configuration, it is advisable to do benchmarks with your application. A rule of thumb is to use the Bi-FM-Index when allowing more than 2 errors.<br>
+This tutorial will show you how to use the seqan3::fm_index and seqan3::bi_fm_index to create indices and how to search them efficiently using seqan3::search.
+
+## Capabilities
+
+With this module you can:
+* Create, store and load (Bi)-FM-Indices
+* Search for exact matches
+* Search for approximate matches (allowing substitutions and indels)
+
+The results of the search can be passed on to other modules, e.g. to create an alignment.
+
+## Terminology
+### Reference
+A reference is the data you want to search in, e.g. a genome or protein database.
+### Query
+A query is the data you want to search for in the reference, e.g. a read or an amino acid sequence.
+### Index
+An index is a data structure usually built over the reference that allows fast searches.
+### FM-Index
+The full-text minute index is a static data structure that allows for fast queries in the indexed text and is a state-of-the-art index. You can find more information on FM-Indicies [in the original publication](https://doi.org/10.1109/SFCS.2000.892127) and [on Wikipedia](https://en.wikipedia.org/wiki/FM-index).
+### Bi-FM-Index
+In contrast to the FM-Index, the Bidirectional Index (Bi-FM-Index) allows searches to be conducted in both directions, i.e. from right to left and left to right. Being able to start at any position of the query in a search improves the search performance drastically when allowing multiple errors. You can find more information on Bi-FM-Indicies [here](https://doi.org/10.1109/BIBM.2009.42).<br>
+However, it also needs to store additional information to be bidirectional. You can think of the Bi-FM-Index as two FM-Indices, one for each direction, hence you need about 2 times more space for the Bi-FM-Index.<br>
+
+## Example
+
+Both FM- and Bi-FM-Indices internally use the implementation of the [Succinct Data Structure Library](https://github.com/simongog/sdsl-lite).<br>
```

Comment:
I would not mention this if people do not need to interface with the sdsl to work with our code.

Reply:
This is probably documented in the API already and links to the API exists as well.